client/landing/stepper/declarative-flow/internals/steps-repository/components/accent-color-control/index.tsx

```diff
@@ -150,7 +167,9 @@ const AccentColorControl = ( { accentColor, setAccentColor }: AccentColorControl
 			<SelectDropdown.Item
 				key={ option.label }
 				icon={ option.icon }
-				onClick={ () => handlePredefinedColorSelect( { value: option.value } ) }
+				onClick={ () =>
+					handlePredefinedColorSelect( { value: option.value, isPremium: option.isPremium } )
```

Comment:
Shouldn't this also check `shouldLimitGlobalStyles` for consistency with how the custom color is tracked?
```suggestion
					handlePredefinedColorSelect( { value: option.value, isPremium: option.isPremium && shouldLimitGlobalStyles } )
```

Reply:
If anything it's the custom color that does an exception and should be consistent with the others! 😄 

Ideally we should only need the color options array to determine whether a color is premium, but the custom color uses a Core component (`ColorPicker`) which is not concerned with our logic, and just expects and returns colors (instead of our "color options").


In https://github.com/Automattic/wp-calypso/pull/70949/commits/baa65de0272ac99d0a1f0734e628971206964b69 I've added a little function that retrieves the custom color's `isPremium` value instead of over-relying on `shouldLimitGlobalStyles`.